### ISSUE TITLE
AddJsonBody in 107.0.1 does not work properly

### ISSUE THREAD

**Issue by sergeisk360** [restsharp/RestSharp#1687]
I have noticed that AddJsonBody  works differently in 107.0.1 version. I am using RestSharp with Pingdom API and have very simple method to pause/resume host. 

Here is an example of method in **106.11.7** version 
```csharp
        public IRestResponse ResumeHost(string hostID)
        {
            var request = new RestRequest("checks", DataFormat.Json);
            request.AddJsonBody(new
            {
                checkids = hostID,
                paused = false
            });
            return client.Put(request);
        }
```

And same code in version **107.0.1**
```csharp
        public async Task<RestResponse> ResumeHostAsync(string hostID)
        {
            var request = new RestRequest("checks") { RequestFormat = DataFormat.Json };
            request.AddJsonBody(new
            {
                checkids = hostID,
                paused = false
            });
            return await client.PutAsync(request);
        }
```

Second code doesn't work properly and looks like object was not posted properly.

**Comment by alexeyzimarev** [comment]
Sorry, but this is not enough

> looks like object was not posted properly.

**Comment by DanWatkins** [comment]
I am trying to upgrade from 107.0.0-preview.17 to 107.0.0-preview.18 and am having trouble with AddJsonBody as well. It seems the body is being encoded when it should not be. See example below. This is also happening on the latest 107.0.2-alpha.0.4.

## Expected Behavior
`{"grant_type":"refresh_token"}`

## Actual Behavior
`"{\u0022grant_type\u0022:\u0022refresh_token\u0022}"`

## Usage
```
string body = "{\"grant_type\":\"refresh_token\"}";
var client = new RestClient(endpoint);
var request = new RestRequest(path, Method.Post);
request.AddJsonBody(bodyString);
client.ExecuteAsync(request);
```

**Comment by alexeyzimarev** [comment]
@DanWatkins  `AddJsonBody` is to add an object, which will be serialized, not a JSON string. If you want to add a JSON string as a body, you need to use `AddBody` and specify the JSON content type.

**Comment by alexeyzimarev** [comment]
@sergeisk360 I can't reproduce the issue. I added tests for it and PUT requests with JSON body work as they should, using plain `RestRequest` and using the extension method.

https://github.com/restsharp/RestSharp/blob/7e3cde4a72f164f68805e30180beb7fc3c1df7b7/test/RestSharp.IntegrationTests/PutTests.cs

**Comment by alexeyzimarev** [comment]
Also this works perfectly fine

https://github.com/restsharp/RestSharp/blob/3e36e84e2e9f5c2ea79c3274172be68f27174d13/test/RestSharp.IntegrationTests/RequestBodyTests.cs#L100-L115

So, I will wait for a couple of days to get more details and then close this issue.

**Comment by milen-denev** [comment]
I want to mention that after I updated to 107.0.1 nothing works anymore. I am still working to find the reason behind that.

**Comment by milen-denev** [comment]
So I am concluding that in my case it simply doesn't add parameters (Using the AddParameter or AddOrUpdateParameter). 
I just nuked the changes and rolled backed the project and everything works fine with 106.15.0.

**Comment by sergeisk360** [comment]
I think this mat relate to ContentType header. In new version **107.0.1.0** the value of it is "application/json; charset=utf-8". In the previous version **106.11.7.0** the value of it is just "application/json". Third party APIs may not accept charset.

**Comment by sergeisk360** [comment]
Adding the following line in code overwrites ContentType ("application/json; charset=utf-8") header 
`request.AddHeader("Content-Type", "application/json");`
and Pingdom API started working properly.

**Comment by alexeyzimarev** [comment]
It's hard to support HTTP servers that don't support the basics of HTTP standards.

If you look at the `StringContent` constructor, you'll see that it sets the encoding unconditionally, wether it's supplied or not.

```csharp
        public StringContent(string content, Encoding? encoding, string? mediaType)
            : base(GetContentByteArray(content, encoding))
        {
            // Initialize the 'Content-Type' header with information provided by parameters.
            MediaTypeHeaderValue headerValue = new MediaTypeHeaderValue((mediaType == null) ? DefaultMediaType : mediaType);
            headerValue.CharSet = (encoding == null) ? HttpContent.DefaultStringEncoding.WebName : encoding.WebName;

            Headers.ContentType = headerValue;
        }
```

**Comment by alexeyzimarev** [comment]
@milen-denev I don't think your issue is related to this one.

>  it simply doesn't add parameters

Of course it does. RestSharp doesn't have 100% test coverage, but the basics are covered and all the tests pass.

**Comment by alexeyzimarev** [comment]
@sergeisk360 I found this article https://gunnarpeipman.com/httpclient-remove-charset/ and implanted an option called `DisableCharset` (it's in client options). By default, it is `false` but if you set it to `true`, you will get no charset in the content type.

**Comment by alexeyzimarev** [comment]
`107.0.2-alpha.0.5` has this option

**Comment by alexeyzimarev** [comment]
I released it in 107.0.2.

**Comment by alexeyzimarev** [comment]
@DanWatkins the latest alpha allows you to add JSON string as body using `AddStringBody(jsonString, DataFormat.Json)` or `AddStringBody(jsonString, "application/json")` (or whatever content type you want). Basically, it's just syntactic wrapper about adding a `BodyParameter`.
